shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetDownloaderServiceImpl.java

```diff
@@ -229,16 +229,30 @@ private void reconvertToNifti(String format, HttpServletResponse response, Long
 		List<String> files = new ArrayList<>();
 		for (File res : workFolder.listFiles()) {
 
+			String datasetFilePath = res.getAbsolutePath();
+			String fileName = res.getName();
+			boolean compressed = datasetFilePath.endsWith(".nii");
+
+			// Gzip file if necessary in order to always return a .nii.gz file
+			if (compressed) {
+				datasetFilePath = datasetFilePath + ".gz";
```

Comment:
Maybe use a constant for the ".gz" zip extension

Reply:
Done!